src/server.c

```diff
@@ -3712,8 +3717,9 @@ void call(client *c, int flags) {
      * since some administrative commands are considered too dangerous to be shown.
      * Other exceptions is a client which is unblocked and retrying to process the command
      * or we are currently in the process of loading AOF. */
-    if (update_command_stats && !reprocessing_command &&
-        !(c->cmd->flags & (CMD_SKIP_MONITOR|CMD_ADMIN))) {
+    if (update_command_stats && !reprocessing_command   &&
+        !(c->cmd->flags & (CMD_SKIP_MONITOR|CMD_ADMIN)) &&
+        commandVisibleForClient(c, real_cmd)) {
```

Comment:
multi-line ifs have brackets in a separate line
```suggestion
        commandVisibleForClient(c, real_cmd))
    {
```